File: project/CCNet.VBDemos.Plugin/Labellers/RandomLabeller.vb
Imports System
Imports Exortech.NetReflector
Imports ThoughtWorks.CruiseControl.Core
Imports ThoughtWorks.CruiseControl.Core.Config

<ReflectorType("randomLabeller")> Public Class RandomLabeller
    Implements ILabeller, IConfigurationValidation

    Public Sub New()
        myMaximumValue = Integer.MaxValue
    End Sub

    Private myMaximumValue As Integer
    <ReflectorProperty("max", Required:=False)> _
    Public Property MaximumValue() As Integer
        Get
            Return myMaximumValue
        End Get
        Set(ByVal value As Integer)
            myMaximumValue = value
        End Set
    End Property

    Public Function Generate(ByVal integrationResult As IIntegrationResult) As String _
    Implements ILabeller.Generate
        Dim rand = New Random()
        Dim label = rand.Next(myMaximumValue).ToString()
        Return label
    End Function

    Public Sub Run(ByVal result As IIntegrationResult) _
    Implements ThoughtWorks.CruiseControl.Core.ITask.Run
        result.Label = Generate(result)
    End Sub

    Public Sub Validate(ByVal configuration As IConfiguration, _
                        ByVal parent As ConfigurationTrace, _
                        ByVal errorProcesser As IConfigurationErrorProcesser) _
                        Implements IConfigurationValidation.Validate
        If myMaximumValue <= 0 Then
            errorProcesser.ProcessError( _
                    "The maximum value must be greater than zero")
        End If
    End Sub
End Class

File: project/CCNet.CSharpDemos.Plugin/Labellers/RandomLabeller.cs
using System;
using Exortech.NetReflector;
using ThoughtWorks.CruiseControl.Core;
using ThoughtWorks.CruiseControl.Core.Config;

namespace CCNet.CSharpDemos.Plugin.Labellers
{
    [ReflectorType("randomLabeller")]
    public class RandomLabeller
        : ILabeller, IConfigurationValidation
    {
        public RandomLabeller()
        {
            this.MaximumValue = int.MaxValue;
        }

        [ReflectorProperty("max", Required = false)]
        public int MaximumValue { get; set; }

        public string Generate(IIntegrationResult integrationResult)
        {
            var rand = new Random();
            var label = rand.Next(this.MaximumValue).ToString();
            return label;
        }

        public void Run(IIntegrationResult result)
        {
            result.Label = this.Generate(result);
        }

        public void Validate(IConfiguration configuration,
            ConfigurationTrace parent,
            IConfigurationErrorProcesser errorProcesser)
        {
            if (this.MaximumValue <= 0)
            {
                errorProcesser.ProcessError(
                    "The maximum value must be greater than zero");
            }
        }
    }
}
